### DISCUSSION TITLE
problems with content-template

### DISCUSSION THREAD

**Discussion by movinator** [dotnet/maui#9428] [General]
Hi,

I'm trying to follow creation of a content-template (manual: ContentView). I have no compile errors, but start of application traps with exception: "StaticResource not found for key Primary".
Primary is the first color of provided resourcedictionary, which i haven't touched.

Could this error happen as some kind of timing errors?
I added the content-template to app.xaml - to be able to use it at various pages.

When I look at the call stack, the exception happens in generated code. Last known code from my project is
App.InitializeComponent();

**Comment by robreno** [comment]
I am running into exact same problem randomly. 

Created basic Maui App with VS2022.
See Resources/Colors.xaml content where colors and static resources are defined:

    <Color x:Key="Primary">#512BD4</Color>
    ...
    <SolidColorBrush x:Key="PrimaryBrush" Color="{StaticResource Primary}"/>
    ...
For some reason the app suddenly stops seeing these defined static resources even though the App.xaml creates the resource dictionary:
 ```
<Application.Resources>
        <ResourceDictionary>
            <ResourceDictionary.MergedDictionaries>
                <ResourceDictionary Source="Resources/Styles/Colors.xaml" />
                <ResourceDictionary Source="Resources/Styles/Styles.xaml" />
            </ResourceDictionary.MergedDictionaries>
        </ResourceDictionary>
    </Application.Resources>
```
If I copy the Color entry with key Primary to the Styles.xaml then the "StaticResource not found for key Primary" goes wasy and it then says "StaticResource not found for key Gray200".

  The Maui.App compiles fine, the error is on deployment when I try to run the app.

When I removed constructor injection on the MainPage the error went away and app deployed and ran fine.
